src/zfile.c

```diff
@@ -100,9 +106,59 @@ zfile_new (const char *path, const char *name)
         }
     }
     zfile_restat (self);
+    self->fd = -1;
+    self->close_fd = false;
     return self;
 }
 
+//  --------------------------------------------------------------------------
+//  Constructor
+//  Create new temporary file for writing via tmpfile. File is automaticaly
+//  deleted on destroy
+
+zfile_t *
+zfile_tmp (void)
+{
+    zfile_t *self = (zfile_t *) zmalloc (sizeof (zfile_t));
+    assert (self);
+    self->remove_on_destroy = true;
+
+    // I know tmpnam is considered insecure, however I did not find a way
+    // how to get fullname from FILE *, or fd, which would be
+    // portable. If someone knows the way, feel free to rewrite it.
+    // ... on non Windows platforms there is O_CREAT | O_EXCL for open,
+    // which prevents the temp file name attacks
+    char name [L_tmpnam];
+    char *r = tmpnam (name);
```

Comment:
Unfortunately this breaks the build on OSX:

```
src/zfile.c:133:15: error: 'tmpnam' is deprecated: This function is provided for compatibility reasons only. Due to security concerns inherent in the design of tmpnam(3), it is highly recommended that you use mkstemp(3) instead. [-Werror,-Wdeprecated-declarations]
 char *r = tmpnam (name);
```

https://travis-ci.org/zeromq/czmq/jobs/292505743

In libzmq we use mkdtemp or mkstemp if that's not available:

https://github.com/zeromq/libzmq/blob/master/src/ipc_listener.cpp#L105

Reply:
An alternative could be to disable that warning with a pragma, at least temporarily

Reply:
It that would be for Linux, it'd be easy. However right now I don't see any easy way ...

Plan to do
* on windows I will keep it unimplemented atm
* if we have F_GETPATH we can use tmpfile, which have almost the same semantics
* if we don't (Linux) then use mkstemp + proc reading